quickwit/quickwit-metastore/src/error.rs

```diff
@@ -118,8 +118,8 @@ impl ServiceError for MetastoreError {
             Self::InvalidManifest { .. } => ServiceErrorCode::Internal,
             Self::Io { .. } => ServiceErrorCode::Internal,
             Self::SourceAlreadyExists { .. } => ServiceErrorCode::BadRequest,
-            Self::SourceDoesNotExist { .. } => ServiceErrorCode::BadRequest,
-            Self::SplitsDoNotExist { .. } => ServiceErrorCode::BadRequest,
+            Self::SourceDoesNotExist { .. } => ServiceErrorCode::NotFound,
+            Self::SplitsDoNotExist { .. } => ServiceErrorCode::NotFound,
```

Comment:
I find that `BadRequest` is better for `SplitsDoNotExist` because it happens when attempting to publish or mark a split for deletion, which is more of an illegal operation than a simple not found error on a CRUD call. This is really the same class of error as `SplitsNotStaged` or `SplitsNotDeletable`.